tests/common/ad-slot.js

```diff
@@ -34,21 +36,46 @@ export class AdSlot {
 		return this.element.getSize();
 	}
 
+	get slotParams() {
+		return this.element.getAttribute('data-gpt-slot-params');
+	}
+
 	get iframeId() {
 		return $(`${this.element} iframe[id^="google_ads_iframe_/5441/"]`).value;
 	}
 
 	get relativeLocationToViewport() {
-		return browser.execute(
-			(givenSelector) => document.querySelector(givenSelector).getBoundingClientRect().y,
-			this.selector,
-		);
+		return helpers.getRelativeLocationToViewport(this.selector);
+	}
+
+	get status() {
+		return {
+			visible: this.element.isDisplayed(),
+			inViewport: this.element.isDisplayedInViewport(),
+			enabled: this.element.isEnabled(),
+		};
+	}
+
+	get height() {
+		return this.element.getSize('height');
+	}
+
+	get width() {
+		return this.element.getSize('width');
 	}
 
 	constructor(config) {
 		this.config = config;
 	}
 
+	getAttribute(attribute) {
+		return this.element.getAttribute(attribute);
+	}
+
+	calculateApectRatio() {
```

Comment:
nitpick: it can be a `getter`
```suggestion
	get aspectRatio() {
```